cub/cub/device/dispatch/dispatch_segmented_sort.cuh

```diff
@@ -1236,10 +1198,8 @@ private:
 
       large_segments_selector.base_segment_offset = current_seg_offset;
       small_segments_selector.base_segment_offset = current_seg_offset;
-      auto current_begin_offset                   = detail::segmented_sort::make_offset_iterator(
-        d_begin_offsets, THRUST_NS_QUALIFIER::constant_iterator<global_segment_offset_t>{current_seg_offset});
-      auto current_end_offset = detail::segmented_sort::make_offset_iterator(
-        d_end_offsets, THRUST_NS_QUALIFIER::constant_iterator<global_segment_offset_t>{current_seg_offset});
+      auto current_begin_offset                   = d_begin_offsets + current_seg_offset;
```

Comment:
I verified that those are indeed random access iterators, so this is great

Reply:
I must be, because the previous code with `offset_iterator` would also add the offset to the iterator using `operator+=`.